subsys/bluetooth/mesh/net.c

```diff
@@ -442,18 +445,26 @@ static int net_header_encode(struct bt_mesh_net_tx *tx, uint8_t nid,
 	return 0;
 }
 
-static int net_encrypt(struct net_buf_simple *buf,
-		       const struct bt_mesh_net_cred *cred, uint32_t iv_index,
-		       bool proxy)
+int bt_mesh_encrypt_and_obfuscate(struct net_buf_simple *buf,
+				  const struct bt_mesh_net_cred *cred,
+				  uint32_t iv_index, bool proxy)
 {
+	uint8_t len = buf->len - NET_PDU_ENC_OFFSET;
 	int err;
 
-	err = bt_mesh_net_encrypt(cred->enc, buf, iv_index, proxy);
+	net_buf_simple_add(buf, BT_MESH_NET_MIC_LEN(buf->data));
+
+	err = bt_mesh_net_encrypt(cred->enc, &buf->data[NET_PDU_ENC_OFFSET], len,
+				  buf->data, &buf->data[NET_PDU_ENC_OFFSET],
+				  BT_MESH_NET_MIC_LEN(buf->data),
+				  iv_index, proxy);
```

Comment:
Could you add error messages in the function like it was in friend.c?
```
BT_ERR("Encrypting failed");
```
```
BT_ERR("Obfuscating failed");
```

Reply:
Done.